protocol_driver_test_fixture.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "protocol_driver_test_fixture.h"
+
+#include "protocol_driver_allocator.h"
+#include "absl/base/internal/sysinfo.h"
+#include "distbench_utils.h"
+#include "glog/logging.h"
+#include "google/protobuf/repeated_field.h"
+
+namespace distbench {
+
+// ProtocolDriver ===============================================
+ProtocolDriverTestFixture::ProtocolDriverTestFixture(int tree_depth) {
+  tree_depth_ = tree_depth;
+}
+
+absl::Status ProtocolDriverTestFixture::Initialize(
+    const ProtocolDriverOptions& pd_opts, int* port) {
+
+  auto pdo = pd_opts;
+  auto server_settings = pdo.mutable_server_settings();
+  google::protobuf::RepeatedPtrField<NamedSetting>::iterator
+      next_protocol_driver_it = server_settings->end();
+  for (auto it=server_settings->begin(); it!=server_settings->end(); it++) {
+      if (it->name()=="driver_under_test") {
+        next_protocol_driver_it = it;
+        pdo.set_protocol_name(it->string_value());
+      }
+  }
+  if (next_protocol_driver_it != server_settings->end())
+    server_settings->erase(next_protocol_driver_it);
+
+  auto maybe_pd_instance_ = AllocateProtocolDriver(pdo, &port_, tree_depth_+1);
```

Comment:
don't declare port_ you can just use the port pointer that is passed in.

Reply:
Done